test/test_torch.py

```diff
@@ -17499,10 +17498,14 @@ def test_fmod(self, device, dtype):
 
         zero = torch.zeros_like(m1)
         if dtype in torch.testing.get_all_int_dtypes():
-            with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
-                m1.fmod(0)
-            with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
-                m1.fmod(zero)
+            if device == 'cpu':
+                with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
+                    m1.fmod(0)
+                with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
+                    m1.fmod(zero)
+            else:
+                self.assertTrue(torch.all(m1.fmod(0) == 4294967295))
+                self.assertTrue(torch.all(m1.fmod(zero) == 4294967295))
```

Comment:
There's also `test_cmod` that is marked as CPU only. I don't know if there's a reason why it was marked that way, but maybe we can make that run on CUDA as well?

Reply:
I am currently trying to do so. 
And I will fix another bug for testing on **CUDA**.
- uint8 `torch.fmod(0) == 255`
- int8/int16/int32 `torch.fmod(0) == -1`
- int64 `torch.fmod(0) == 4294967295`

Reply:
I will remove `test_cmod`. The tests for correctness of `torch.fmod` against `math.fmod` will be moved into `test_fmod`.